src/libsync/creds/keychainchunk.cpp

```diff
@@ -61,11 +69,45 @@ WriteJob::WriteJob(Account *account, const QString &key, const QByteArray &data,
     _chunkCount = 0;
 }
 
+WriteJob::WriteJob(const QString &key, const QByteArray &data, QObject *parent)
+    : WriteJob(nullptr, key, data, parent)
+{
+#ifdef Q_OS_WIN
+    // NOTE: The following is normally done in AbstractCredentials::keychainKey
+    //       when an _account is specified by our other ctr overload (see 'kck' in this file).
+
+    // On Windows the credential keys aren't namespaced properly
+    // by qtkeychain. To work around that we manually add namespacing
+    // to the generated keys. See #6125.
+    // It's safe to do that since the key format is changing for 2.4
+    // anyway to include the account ids. That means old keys can be
+    // migrated to new namespaced keys on windows for 2.4.
+    _key.prepend(QCoreApplication::applicationName() + "_");
+#endif
+}
+
 void WriteJob::start()
 {
+    _isJobRunning = true;
     slotWriteJobDone(nullptr);
 }
 
+bool WriteJob::startAwait()
+{
+    start();
+
+    while (_isJobRunning) {
```

Comment:
A better pattern would be to call `exec()` on a `QEventLoop` (stack allocated) and connect the `exit()` slot of that event loop to the `finished()` signal of this.

This was no extra `_isJobRunning` state to track (which tends to be error prone).

Reply:
Good point, I was wondering if there wouldn't have been such a nice solution available for `ProxyAuthHandler`. I will adapt it there too then :)